src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    }
 
-// Viking
-class Viking {}
+    attack() {
+     return this.strength
+    }
+    receiveDamage(damage) {
+     this.health -= damage   
+    }
+}
 
-// Saxon
-class Saxon {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name
+    }
+    receiveDamage(damage) {
+        this.health -= damage
+        let dead = this.health < 0
+        if (Viking !== dead) {
+            console.log(`${Viking.name} has received ${damage} points of damage`)
+        } else {
+            console.log(`${Viking.name} has died in act of combat`)
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+}
+
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        Saxon.health -= damage
+        let dead = this.health < 0
+        if (Saxon === dead) {
+            console.log('A Saxon has died in combat')
+        } else {
+            console.log(`A Saxon has received ${damage} points of damage`)
+        }
+
+    attack() {
+        return Saxon.strength
+    }
+}
+}
 
 // War
-class War {}
+class War {
+    constructor (vikingArmy, saxonArmy)
+    this.vikingArmy = []
+    this.saxonArmy = []
+
+    function addViking(this.Viking) {
+        return this.Viking + 1
+    }
+
+    function addSaxon(this.Saxon) {
+        return this.Saxon + 1
+    }
+
+    function vikingAttack() {
```

Comment:
Anne, retire primeiramente todas essas 'function' dessa última class War. 
No método vikingAttack( ), precisamos dizer qual viking que é, qual saxon que é, e se o saxon sofre apenas um dano ou morre. 
Para isso, vamos pegar de maneira aleatória qual index que é o viking: 
let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length)
Em seguida, vamos identificar esse viking no exercito viking, vikinArmy: 
let randomViking = this.vikingArmy[randomVikingIndex] 
Agora, você vai fazer isso também para o saxons, com o mesmo raciocínio e lógica acima, que foi feito para os vikings. Ok? 
Depois que vc fizer isso, aí veremos a luta e o resultado dela: 
let fight = randomSaxon.receiveDamage(randomViking.attack())
        if(randomSaxon.health <= 0){
            this.saxonArmy.splice(randomSaxonIndex, 1)
        }
Como é um ataque contra um saxon aqui nesse método, verificamos se ele continua e retornamos a luta no método:
return figth
} e aqui termina o método